### PR TITLE
♻️ Refactor code on ImageData, ImageHandler and Integrate OpenCV Java

### PR DESCRIPTION
**Pull request by ken479**
# Description

Integrated OpenCV Java into Classifai to support webp image and other image types. Refactor some codes from ImageHandler and make ImageData as abstract class to provide a layer of screening on Image types and make ImageData more organized to be used in ImageHandler. 

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (Non-breaking change which fixes an issue)
- [ ] New feature (Non-breaking change which adds functionality)
- [ ] Breaking change (Fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Example: REST API changes)

# Tested on?

- [x] Windows  
- [x] Linux Ubuntu 
- [ ] Centos 
- [ ] Mac  
- [ ] Others  (State here -> xxx )  

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged
- [ ] CHANGELOG.md has been updated
- [ ] Update [Gitbook REST API Page](https://docs.classifai.ai/v/dev/development/rest-api)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -257,6 +257,11 @@
                 <artifactId>metadata-extractor</artifactId>
                 <version>${metadata.extractor.version}</version>
             </dependency>
+            <dependency>
+                <groupId>org.openpnp</groupId>
+                <artifactId>opencv</artifactId>
+                <version>4.5.1-2</version>
```
Comment (codenamewei) [inline pom.xml:263]: dependency define in [properties](https://github.com/CertifaiAI/classifai/blob/08cafc2bb72e728290182a7f65fb6605d2f8f567/pom.xml#L72-L101)
and call 
it to here example
```
<dependency>
                <groupId>org.openpnp</groupId>
                <artifactId>opencv</artifactId>
                <version>{opencv.version}</version>
</dependency>
```

File: classifai-core/pom.xml [hunk 2]
```diff
@@ -116,6 +116,11 @@
             <groupId>com.drewnoakes</groupId>
             <artifactId>metadata-extractor</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.openpnp</groupId>
+            <artifactId>opencv</artifactId>
+            <version>4.5.1-2</version>
```
Comment (codenamewei) [inline classifai-core/pom.xml:122]: in submodule pom.xml no need to define version
it will find back the version in the root pom.xml

**Review by codenamewei — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (ken479) [reply pom.xml:263]: Thank you @codenamewei for the suggestion

File: classifai-core/pom.xml (hunk 2 above)
Comment (ken479) [reply classifai-core/pom.xml:122]: Noted. will remove it.

File: classifai-core/src/main/java/ai/classifai/data/type/image/ImageData.java [hunk 3]
```diff
@@ -15,23 +15,111 @@
...
+        public static ImageData getImageData(Metadata metadata) throws NotSupportedImageTypeException {
+            if (metadata.containsDirectoryOfType(JpegDirectory.class)) {
+                return new JpegImageData(metadata);
+            } else if (metadata.containsDirectoryOfType(PngDirectory.class)) {
+                return new PngImageData(metadata);
+            } else if (metadata.containsDirectoryOfType(BmpHeaderDirectory.class)) {
+                return new BmpImageData(metadata);
+            } else if (metadata.containsDirectoryOfType(WebpDirectory.class)) {
+                return new WebpImageData(metadata);
+            } else {
```
Comment (YCCertifai) [inline classifai-core/src/main/java/ai/classifai/data/type/image/ImageData.java:118]: need to screen out animated webp images from here too, currently animated webp image will be allowed, and create errors in application.
image

File: classifai-core/src/main/java/ai/classifai/database/annotation/AnnotationVerticle.java [hunk 4]
```diff
@@ -481,21 +481,21 @@ public void queryData(Message<JsonObject> message, @NonNull String annotationKey
             BufferedImage img = WasabiImageHandler.getThumbNail(loader.getWasabiProject(), annotation.getImgPath());
 
             //not checking orientation for on cloud version
-            imgData = ImageHandler.getThumbNail(img, false, null);
+//            imgData = ImageHandler.getThumbNail(img);
```
Comment (YCCertifai) [inline classifai-core/src/main/java/ai/classifai/database/annotation/AnnotationVerticle.java:484]: need to remove the commented unused code, unless there is a special purpose for it

**Review by YCCertifai — CHANGES_REQUESTED** [review]

**Pull request closed without merging**